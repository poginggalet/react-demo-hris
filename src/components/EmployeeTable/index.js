import React from 'react';


const EmployeeTable = (({personalDetails, jobDetails, benefitsDetails, onUpdateForm}) => {
        return (
            <div>
            <div>
                <h3> PERSONAL DETAILS</h3>
                <ul>
                    <li>First Name {personalDetails.firstName} </li>
                    <li> Middle Name {personalDetails.middleName}</li>
                    <li> Last Name {personalDetails.lastName}</li>
                    <li> Gender {personalDetails.gender}</li>
                    <li> Title {personalDetails.title}</li>
                    <li> Address {`${personalDetails.address.unitNumber} ${personalDetails.address.street} ${personalDetails.address.city} ${personalDetails.address.province} ${personalDetails.address.region} ${personalDetails.address.zipCode}`}</li>}
                </ul>
            </div>
             <div>
                <h3> CONTACT DETAILS</h3>
                <ul>
                    <li> Landline {personalDetails.contact.landlineNumber}</li>
                    <li> Mobile Number {personalDetails.contact.mobileNumber}</li>
                    <li> Email {personalDetails.contact.email}</li>
                </ul>
            </div>
            <div>
                <h3> JOB DETAILS</h3>
                <ul>
                    <li> Title {jobDetails.title}</li>
                    <li> Employee Number {jobDetails.employeeNumber}</li>
                    <li> Location {jobDetails.location}</li>
                    <li> Department {jobDetails.department}</li>
                    <li> Salary {jobDetails.salary}</li>
                </ul>
            </div>
            <div>
                <h3> EMPLOYEE BENEFITS </h3>
                <ul>
                    <li> SSS {benefitsDetails.SSS}</li>
                    <li> PhilHealth{benefitsDetails.Philhealth}</li>
                    <li> PAGIBIG {benefitsDetails.PAGIBIG}</li>
                    <li> BIR {benefitsDetails.BIR}</li>
                </ul>
            </div>
            <button onClick= {onUpdateForm}> Update </button>
        </div>

        );
})

export { EmployeeTable };