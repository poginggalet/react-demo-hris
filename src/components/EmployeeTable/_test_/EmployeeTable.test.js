import React from 'react';
import Enzyme, { render, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { createSerializer } from "enzyme-to-json";
import { EmployeeTable }  from '../../EmployeeTable';

expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));
Enzyme.configure({ adapter: new Adapter() });

describe('Employee Table Component test', () => {
    it('should simulate click event on material-ui Delete IconButton', () => {
        const onUpdateForm = jest.fn();
        const detail = {
            personalDetails: {
                firstName: "Diana",
                lastName: "Cuerto",
                middleName: "Filamina",
                gender: "Female",
                title: "Ms",
                address: {
                    unitNumber: "3-F",
                    street: "Alley 28",
                    city: "Quezon City",
                    province: "Manila",
                    region: "NCR",
                    zipCode: 1100
                },
                contact: {
                    landlineNumber: "455-42-52",
                    mobileNumber: "09173942673",
                    email: "dianca_cueto@gmail.com"
                }
            },
            jobDetails: {
                title: "Software Engineer",
                employeeNumber: "EMP-1002",
                location: "Buendia Branch",
                department: "Engineering Department",
                email: "diana_cueto@pearlpay.com",
                salary: 35000
            },
            benefitsDetails: {
                SSS: "0407514490",
                Philhealth: "02050905813",
                PAGIBIG: "123456789101",
                BIR: "265683857000"
            }
        }
        const wrapper = shallow(<EmployeeTable
            personalDetails = {detail.personalDetails}
            jobDetails = {detail.jobDetails}
            benefitsDetails = {detail.benefitsDetails}
            onUpdateForm={onUpdateForm}/>);
        wrapper.find('button').simulate("click");
        wrapper.find('button').simulate("click");
        expect(onUpdateForm.mock.calls.length).toEqual(2);
    })
    it("renders Employee Table Component", () => {
        const detail = {
                personalDetails: {
                    firstName: "Diana",
                    lastName: "Cuerto",
                    middleName: "Filamina",
                    gender: "Female",
                    title: "Ms",
                    address: {
                        unitNumber: "3-F",
                        street: "Alley 28",
                        city: "Quezon City",
                        province: "Manila",
                        region: "NCR",
                        zipCode: 1100
                    },
                    contact: {
                        landlineNumber: "455-42-52",
                        mobileNumber: "09173942673",
                        email: "dianca_cueto@gmail.com"
                    }
                },
                jobDetails: {
                    title: "Software Engineer",
                    employeeNumber: "EMP-1002",
                    location: "Buendia Branch",
                    department: "Engineering Department",
                    email: "diana_cueto@pearlpay.com",
                    salary: 35000
                },
                benefitsDetails: {
                    SSS: "0407514490",
                    Philhealth: "02050905813",
                    PAGIBIG: "123456789101",
                    BIR: "265683857000"
                }
            }
        const wrapper = render(
            <EmployeeTable
                personalDetails = {detail.personalDetails}
                jobDetails = {detail.jobDetails}
                benefitsDetails = {detail.benefitsDetails}/>
        );
        expect(wrapper).toMatchSnapshot();
    });
});