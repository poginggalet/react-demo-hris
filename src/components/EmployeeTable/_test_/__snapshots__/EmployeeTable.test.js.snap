// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employee Table Component test renders Employee Table Component 1`] = `
<div>
  <div>
    <h3>
       PERSONAL DETAILS
    </h3>
    <ul>
      <li>
        First Name Diana 
      </li>
      <li>
         Middle Name Filamina
      </li>
      <li>
         Last Name Cuerto
      </li>
      <li>
         Gender Female
      </li>
      <li>
         Title Ms
      </li>
      <li>
         Address 3-F Alley 28 Quezon City Manila NCR 1100 
      </li>
    </ul>
  </div>
  <div>
    <h3>
       CONTACT DETAILS
    </h3>
    <ul>
      <li>
         Landline 455-42-52
      </li>
      <li>
         Mobile Number 09173942673
      </li>
      <li>
         Email dianca_cueto@gmail.com
      </li>
    </ul>
  </div>
  <div>
    <h3>
       JOB DETAILS
    </h3>
    <ul>
      <li>
         Title Software Engineer
      </li>
      <li>
         Employee Number EMP-1002
      </li>
      <li>
         Location Buendia Branch
      </li>
      <li>
         Department Engineering Department
      </li>
      <li>
         Salary 35000
      </li>
    </ul>
  </div>
  <div>
    <h3>
       EMPLOYEE BENEFITS 
    </h3>
    <ul>
      <li>
         SSS 0407514490
      </li>
      <li>
         PhilHealth02050905813
      </li>
      <li>
         PAGIBIG 123456789101
      </li>
      <li>
         BIR 265683857000
      </li>
    </ul>
  </div>
  <button>
     Update 
  </button>
</div>
`;
